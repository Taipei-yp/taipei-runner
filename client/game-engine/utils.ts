import { CollisionBox } from "./models";

/** Получение изменения X координаты */
export function getIncrement(deltaTime: number, speed: number, fps: number) {
  return Math.floor(speed * (fps / 1000) * deltaTime);
}

/**
 * Получение случайного значения в заданных границах
 */
export function getRandomNum(min: number, max: number): number {
  return Math.floor(Math.random() * (max - min + 1)) + min;
}

/**
 * Получение текущего времени.
 */
export function getTimeStamp(): number {
  if (window.performance && window.performance.now)
    return window.performance.now();
  return new Date().getTime();
}

/**
 * Создание поля пересечения относительно другого.
 * Если координата поля отсчета отрицательное значение отсчет идет от правого края поля отсчета
 * @param box Поле пересечения
 * @param relative Поле отсчета
 * @param scf Коэффициент изменения размера
 */
export function createRelativeCollisionBox(
  box: CollisionBox,
  relative: CollisionBox,
  scf = 1,
): CollisionBox {
  return new CollisionBox(
    relative.x + box.x * scf + (box.x < 0 ? relative.width : 0),
    relative.y + box.y * scf + (box.y < 0 ? relative.height : 0),
    box.width * scf,
    box.height * scf,
  );
}

/**
 * Отрисовка областей пересечения при отладке
 */
export function drawCollisionBoxes(
  canvasCtx: CanvasRenderingContext2D,
  heroBox: CollisionBox,
  obstacleBox: CollisionBox,
) {
  canvasCtx.save();
  // eslint-disable-next-line no-param-reassign
  canvasCtx.strokeStyle = "#f00";
  canvasCtx.strokeRect(heroBox.x, heroBox.y, heroBox.width, heroBox.height);

  // eslint-disable-next-line no-param-reassign
  canvasCtx.strokeStyle = "#0f0";
  canvasCtx.strokeRect(
    obstacleBox.x,
    obstacleBox.y,
    obstacleBox.width,
    obstacleBox.height,
  );
  canvasCtx.restore();
}

/**
 * Пересечение двух областей
 * @param heroBox Область пересечения героя
 * @param obstacleBox Область пересечения препятствия
 * @return {boolean} Пересекаются да/нет
 */
export function boxCompare(
  heroBox: CollisionBox,
  obstacleBox: CollisionBox,
): boolean {
  return (
    heroBox.x < obstacleBox.x + obstacleBox.width &&
    heroBox.x + heroBox.width > obstacleBox.x &&
    heroBox.y < obstacleBox.y + obstacleBox.height &&
    heroBox.height + heroBox.y > obstacleBox.y
  );
}
