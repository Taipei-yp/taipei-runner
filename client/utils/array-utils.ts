export const arrayOfNumbersInRange = (from: number, to: number): number[] => {
  const result = [];
  for (let i = from; i <= to; ++i) {
    result.push(i);
  }
  return result;
};
