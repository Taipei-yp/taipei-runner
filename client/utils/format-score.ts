const NUM_LENGTH = 6;

export function formatScore(score: number) {
  return score.toString().padStart(NUM_LENGTH, "0");
}
