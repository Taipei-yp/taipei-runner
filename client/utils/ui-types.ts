export type UserInfo = {
  avatar: string;
  name: string;
};

export type TableSort = {
  field: string | null;
  direction: SortDirection | null;
};

export type SortDirection = "asc" | "desc";

export type SortRequestHandler = (field: string) => void;
