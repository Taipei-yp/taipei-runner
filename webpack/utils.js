const path = require("path");
const childProcess = require("child_process");

const IS_DEV = process.env.NODE_ENV !== "production";

function rootDir(extraPath = "") {
  return path.resolve(__dirname, "../", extraPath);
}

function srcDir(extraPath = "") {
  return rootDir(`./client${extraPath}`);
}

function distDir(extraPath = "") {
  return rootDir(`./dist${extraPath}`);
}

function getAppVersion() {
  // eslint-disable-next-line global-require
  const pkg = require("../package.json");
  return pkg.version;
}

function getBranchName() {
  return childProcess
    .execSync("git rev-parse --abbrev-ref HEAD")
    .toString()
    .trim();
}

function getCommitId() {
  return childProcess.execSync("git rev-parse HEAD").toString().trim();
}

function getBuildDate() {
  return Date.now();
}

function getBuildInfo() {
  const appVersion = getAppVersion();
  const branchName = getBranchName();
  const commitId = getCommitId();
  const buildDate = getBuildDate();
  return {
    buildDate,
    version: appVersion,
    branchName,
    commitId,
  };
}

module.exports = {
  IS_DEV,
  rootDir,
  srcDir,
  distDir,
  getBuildInfo,
};
