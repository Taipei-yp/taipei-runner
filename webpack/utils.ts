import childProcess from "child_process";
import path from "path";
import pkg from "../package.json";

export const IS_DEV = process.env.NODE_ENV !== "production";

export function rootDir(extraPath = "") {
  return path.resolve(__dirname, "../", extraPath);
}

export function srcDir(extraPath = "") {
  return rootDir(`./client${extraPath}`);
}

export function distDir(extraPath = "") {
  return rootDir(`./dist${extraPath}`);
}

function getAppVersion() {
  return pkg.version;
}

function getBranchName() {
  return childProcess
    .execSync("git rev-parse --abbrev-ref HEAD")
    .toString()
    .trim();
}

function getCommitId() {
  return childProcess.execSync("git rev-parse HEAD").toString().trim();
}

function getBuildDate() {
  return Date.now();
}

export function getBuildInfo() {
  const appVersion = getAppVersion();
  const branchName = getBranchName();
  const commitId = getCommitId();
  const buildDate = getBuildDate();
  return {
    buildDate,
    version: appVersion,
    branchName,
    commitId,
  };
}

export function clientJsFileName(isDev: boolean, pathDataChunkName?: string) {
  let name = "[name].js";
  if (!isDev && pathDataChunkName !== undefined) {
    if (pathDataChunkName === "sw") {
      name =
        pathDataChunkName === "sw" ? "sw.js" : "js/[name].[contenthash].js";
    }
  }
  return name;
}
