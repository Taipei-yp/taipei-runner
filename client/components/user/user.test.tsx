import React from "react";
import TestRenderer from "react-test-renderer";
import { UserInfo } from "client/utils/ui-types";
import { User } from ".";

describe("Component User", () => {
  const testUser: UserInfo = {
    name: "testname",
    avatar: "/testsrc",
  };
  const user = TestRenderer.create(<User value={testUser} />);

  it("correct render", () => {
    expect(user.toJSON()).toMatchSnapshot();
  });
});
