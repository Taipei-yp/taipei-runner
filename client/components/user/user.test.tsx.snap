// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component User correct render 1`] = `
<div
  className="user"
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <img
      alt=""
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
      src="/testsrc"
    />
  </div>
  <span
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    testname
  </span>
</div>
`;
